camayoc/tests/qpc/ui/test_credentials.py

```diff
@@ -35,22 +69,46 @@ def test_create_delete_credential(selenium_browser, qpc_login, source_type):
         4) Delete the newly created credential.
     :expectedresults: A new credential is created and then deleted.
     """
-    name = utils.uuid4()
-    username = utils.uuid4()
-    password = utils.uuid4()
-    create_credential(
-        selenium_browser,
-        {
-            "name": name,
-            "username": username,
-            "password": password,
-            "source_type": source_type,
-        },
+    cleanup.append(Credential(name=credential.credential_form_dto.credential_name))
+    (
+        ui_client.begin()
+        .login(data_factories.LoginFormDTOFactory())
+        .navigate_to(MainMenuPages.CREDENTIALS)
+        .add_credential(credential)
+        .logout()
+    )
+
+
+def network_credentials_provider():
```

Comment:
Personally I would have supporting functions at the top and then all tests together below, but I guess it's a valid option to have supporting function closer to where it is used.

Reply:
Let's adopt utilities functions at the top.